### ISSUE TITLE
compliance: §6.7.9/6.7.11/6.7.3 Annotation AP stream violations not detected — 20 false negatives

### ISSUE THREAD

**Issue by jasperdew** [jasperdew/xfa-native-rust#454]
## Summary
The compliance checker misses several **annotation-related** ISO 19005-2 violations. 20+ PDFs from the veraPDF suite and stressful corpus pass our checker but fail veraPDF.

## Impact
~20 false negatives in 4K corpus (run-20260315-065448.905)

## Failing PDFs by clause
```
§6.7.9 (Annotation AP streams required for non-invisible, non-hidden annotations):
  tagged/verapdf/isartor-6-7-2-t02-fail-c.pdf
  tagged/verapdf/isartor-6-7-9-t01-fail-a.pdf
  tagged/verapdf/veraPDF test suite 6-7-2-t03-fail-o.pdf
  tagged/verapdf/veraPDF test suite 6-7-2-t06-fail-c.pdf
  tagged/verapdf/veraPDF test suite 6-7-2-t08-fail-d.pdf
  tagged/verapdf/veraPDF test suite 6-7-2-t09-fail-b/i/l.pdf
  tagged/verapdf/veraPDF test suite 6-7-2-t14-fail-n.pdf
  tagged/verapdf/veraPDF test suite 6-7-2-t15-fail-d/w.pdf
  tagged/verapdf/veraPDF test suite 6-7-2-t17-fail-g/k/x.pdf
  stressful/batch1/PDFBOX/PDFBOX-1760-11.pdf
  stressful/batch5/PDFIUM/PDFIUM-610-0.pdf
  stressful/batch5/poppler/poppler-106863-0.pdf

§6.7.11 (Annotation /CA must be 1.0):
  tagged/verapdf/isartor-6-7-9-t01-fail-a.pdf
  tagged/verapdf/veraPDF test suite 6-7-11-t05-fail-a.pdf
  tagged/verapdf/veraPDF test suite 6-6-4-t01-fail-f.pdf
  stressful/batch1/PDFBOX/PDFBOX-3017-0.pdf
  stressful/batch5/PDFIUM/PDFIUM-610-0.pdf
  stressful/batch5/poppler/poppler-106863-0.pdf

§6.7.3 (Annotation /F flag — Print bit must be set; Hidden/NoView bits must not):
  stressful/batch1/PDFBOX/PDFBOX-1760-11.pdf
  stressful/batch5/PDFIUM/PDFIUM-610-0.pdf

§6.7.2.1 (Annotation /Contents must be a string):
  tagged/verapdf/veraPDF test suite 6-7-2-1-t01-fail-b.pdf
```

## Spec
- **§6.7.9**: Annotations except Popup must have `/AP` dict with `/N` entry unless `/F` flags include Invisible or NoView
- **§6.7.11**: Annotation `/CA` (constant opacity) must be 1.0
- **§6.7.3**: `/F` Print bit (bit 3) must be set; Hidden (bit 1) and NoView (bit 8) must not be set
- **§6.7.2.1**: `/Contents` must be a text string

## Fix location
`crates/pdf-compliance/src/check.rs` — §6.7 annotation checks.
